src/main/java/org/prebid/server/bidder/adgeneration/AdgenerationBidder.java

```diff
@@ -113,13 +115,17 @@ private String getUri(String endpointUrl, String adSize, String id, String curre
                 .addParameter("adapterver", VERSION);
 
         if (StringUtils.isNotBlank(adSize)) {
-            uriBuilder.addParameter("size", adSize);
+            uriBuilder.addParameter("sizes", adSize);
         }
 
         if (site != null && StringUtils.isNotBlank(site.getPage())) {
             uriBuilder.addParameter("tp", site.getPage());
         }
 
+        if (source != null && StringUtils.isNotBlank(source.getTid())) {
```

Comment:
Minor. It may looks simpler if we extract needed param to separate variable, like:
```
sourceTid = source != null ? source.getTid() : null;
if (StringUtils.isNotEmpty(sourceTid))
{
    //...
}
```
The same can be done for `site.getPage()` in case above and `device.getUa()` below.